compiler/rustc_driver/src/lib.rs

```diff
@@ -1317,7 +1329,14 @@ pub fn main() -> ! {
                 })
             })
             .collect::<Vec<_>>();
-        RunCompiler::new(&args, &mut callbacks).run()
+        match process_args(&args) {
+            Some(matches) => {
+                let sopts = config::build_session_options(&matches);
+                init_rustc_env_logger(sopts.debugging_opts.always_color_logs);
+                RunCompiler::new(matches, sopts, &mut callbacks).run()
```

Comment:
I guess the alternative would be to use an env var instead of a `-Z` flag so that we don't have to parse the options quite so early. In particular, are we sure we don't need the logger set up already when parsing arguments? Looks to me like it was deliberately done so early.

Reply:
It was deliberately early, and if we want to have logging while parsing arguments, then we do need an env var indeed. I don't have a strong opinion here. Though I guess we should probably either use an env var or get some more compiler team opinions in here.

Reply:
An env var sounds fine to me, then we don't have to worry about stable/unstable either.

Reply:
Oh, that would be so much easier to implement! Thanks for suggesting that @RalfJung :)